### PR TITLE
US129172 - Allow deselection

### PR DESCRIPTION
**Pull request by devpow112**
Did this off your since I used `selectLevelAt`. Tested with master merged as well and this works with the removal of active state stuff.

### CODE REVIEW

File: d2l-rubric-levels-mobile.js [hunk 1]
```diff
@@ -395,13 +395,25 @@ Polymer({
...
-		if (this.focused === level.dataIndex
-		&& this.selected === level.dataIndex) {
+		if (this.focused === level.dataIndex &&
+			this.selected === level.dataIndex) {
 			return;
 		}
-		this._focusLevel(level.dataIndex);
+
+		if (this.selected === level.dataIndex) {
+			this.selected = undefined;
```
Comment (semibran) [inline d2l-rubric-levels-mobile.js:404]: I've typically been setting selected/hovered/focused to -1 when they're not in use, though I think the default value is undefined so this probably doesn't break anything

File: d2l-rubric-levels-mobile.js [hunk 2]
```diff
@@ -395,13 +395,25 @@ Polymer({
...
+		}
+
+		this.focusSlider();
+
 		if (!this.readOnly) {
-			this._getCriterion().selectCell(() => this.cellAssessmentMap[level.dataset.cellHref]);
+			const levelCellHref = level.dataset.cellHref;
+			const cellAssessment = this.cellAssessmentMap[levelCellHref];
+
+			this._getCriterion().selectCell(() => cellAssessment);
```
Comment (semibran) [inline d2l-rubric-levels-mobile.js:416]: I'd be careful about streamlining the entity getter like this..... it looks like the cell assessment map is subject to change for some reason, so I've run into some issues with this kind of thing while working with selection logic before. Maybe you're a bit more familiar than me about why that might be the case.

File: d2l-rubric-levels-mobile.js [hunk 3]
```diff
@@ -451,7 +463,9 @@ Polymer({
 	},
 
 	_handleThumbClick: function() {
-		this.focusSlider();
+		if (this.selected >= 0) {
+			this.selectLevelAt(this.selected);
+		}
```
Comment (semibran) [inline d2l-rubric-levels-mobile.js:468]: May or may not conflict with #823

File: d2l-rubric-levels-mobile.js (hunk 2 above)
Comment (devpow112) [reply d2l-rubric-levels-mobile.js:416]: K will revert

File: d2l-rubric-levels-mobile.js (hunk 1 above)
Comment (devpow112) [reply d2l-rubric-levels-mobile.js:404]: This worked in my testing. Let me try `-1` and see how it goes.

**Comment by devpow112** [comment]
Tests timed out. I'm gonna increase our test timeout

File: d2l-rubric-levels-mobile.js (hunk 1 above)
Comment (devpow112) [reply d2l-rubric-levels-mobile.js]: Seemed to work the same with `-1` but feels safer. Changed.

File: d2l-rubric-levels-mobile.js (hunk 2 above)
Comment (devpow112) [reply d2l-rubric-levels-mobile.js]: Reverted

**Review by semibran — APPROVED** [review]

**Pull request merged**